crates/workspace/src/pane_group.rs

```diff
@@ -1183,12 +1225,16 @@ mod element {
                             .apply_along(Axis::Horizontal, |val| val - Pixels(1.)),
                     };
 
-                    if overlay_opacity.is_some() && self.active_pane_ix != Some(ix) {
+                    if overlay_opacity.is_some()
+                        && child.is_leaf_pane
+                        && self.active_pane_ix != Some(ix)
+                    {
                         window.paint_quad(gpui::fill(overlay_bounds, overlay_background));
                     }
 
                     if let Some(border) = overlay_border {
-                        if self.active_pane_ix == Some(ix) {
+                        // Only draw borders on active leaf panes
```

Comment:
Let's not add comments that explain a single line of code below, the `if` statement below is pretty self-documenting (after all, the `if` statement above is not documented).